In this notebook I am exploring the relations between various factors such as Market Value, Age, Position, Clubs.

In [2]:
#Importing Libraries
import pandas as pd
import numpy as np
from plotly.offline import plot, iplot, init_notebook_mode
import plotly.graph_objs as go

# Make plotly work with Jupyter notebook
init_notebook_mode(connected = True)

In [3]:
#Importing dataset
epl = pd.read_csv('epldata_final.csv')

Lets get an overview of the data.

In [93]:
epl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 461 entries, 0 to 460
Data columns (total 17 columns):
name            461 non-null object
club            461 non-null object
age             461 non-null int64
position        461 non-null object
position_cat    461 non-null int64
market_value    461 non-null float64
page_views      461 non-null int64
fpl_value       461 non-null float64
fpl_sel         461 non-null object
fpl_points      461 non-null int64
region          460 non-null float64
nationality     461 non-null object
new_foreign     461 non-null int64
age_cat         461 non-null int64
club_id         461 non-null int64
big_club        461 non-null int64
new_signing     461 non-null int64
dtypes: float64(3), int64(9), object(5)
memory usage: 61.3+ KB


In [94]:
epl['club'].describe()

count         461
unique         20
top       Everton
freq           28
Name: club, dtype: object

So this dataset contains 461 players from 20 different clubs. The clubs are

In [95]:
clubs = tuple(set(epl['club']))
print(clubs)

('Bournemouth', 'Liverpool', 'Leicester+City', 'Burnley', 'Huddersfield', 'Crystal+Palace', 'Brighton+and+Hove', 'Newcastle+United', 'Stoke+City', 'Tottenham', 'Swansea', 'Manchester+United', 'West+Ham', 'Southampton', 'Watford', 'West+Brom', 'Arsenal', 'Manchester+City', 'Everton', 'Chelsea')


In [26]:
value = []
for club in clubs:
    value.append(sum(epl['market_value'].loc[epl['club']==club]))

In [ ]:
Now lets visualise some of these data

In [38]:
keys= clubs
values=value

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Market Value of players")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [40]:
average_age = []
for club in clubs:
    average_age.append(np.mean(epl['age'].loc[epl['club']==club]))

In [97]:
keys= clubs
values=average_age

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Average Age")
})

PlotlyDictKeyError: 'color' is not allowed in 'bar'

Path To Error: ['color']

Valid attributes for 'bar' at path [] under parents []:

    ['bardir', 'base', 'basesrc', 'customdata', 'customdatasrc', 'dx',
    'dy', 'error_x', 'error_y', 'hoverinfo', 'hoverinfosrc', 'hoverlabel',
    'hovertext', 'hovertextsrc', 'ids', 'idssrc', 'insidetextfont',
    'legendgroup', 'marker', 'name', 'offset', 'offsetsrc', 'opacity',
    'orientation', 'outsidetextfont', 'r', 'rsrc', 'showlegend', 'stream',
    't', 'text', 'textfont', 'textposition', 'textpositionsrc', 'textsrc',
    'tsrc', 'type', 'uid', 'visible', 'width', 'widthsrc', 'x', 'x0',
    'xaxis', 'xcalendar', 'xsrc', 'y', 'y0', 'yaxis', 'ycalendar', 'ysrc']

Run `<bar-object>.help('attribute')` on any of the above.
'<bar-object>' is the object at []

In [49]:
country, counts = np.unique(epl['nationality'], return_counts=True)

In [52]:
keys= country
values=counts

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Nationality of Players")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [53]:
c_value = []
for c in country:
    c_value.append(sum(epl['market_value'].loc[epl['nationality']==c]))

In [66]:
keys= country
values=c_value

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Market Value vs Nationality")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [68]:
c_value_mean = []
for c,v in zip(c_value,counts):
    c_value_mean.append(c/v)

In [69]:
keys= country
values=c_value_mean

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Mean Market Value vs Nationality")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [70]:
c_value_median = []
for c in country:
    c_value_median.append(np.median(epl['market_value'].loc[epl['nationality']==c]))

In [71]:
keys= country
values=c_value_median

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Market Value vs Nationality")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [72]:
c_value_max = []
for c in country:
    c_value_max.append(max(epl['market_value'].loc[epl['nationality']==c]))

In [73]:
keys= country
values=c_value_max

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Market Value vs Nationality")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [81]:
ages = tuple(set(epl['age']))
age_value_mean = []
age_value_median = []
age_value_max = []
for age in ages:
    age_value_mean.append(np.mean(epl['market_value'].loc[epl['age']==age]))
    age_value_median.append(np.median(epl['market_value'].loc[epl['age']==age]))
    age_value_max.append(max(epl['market_value'].loc[epl['age']==age]))

In [77]:
keys= ages
values=age_value_mean

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Market Value vs Age")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [78]:
keys= ages
values=age_value_median

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Market Value vs Age")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [82]:
keys= ages
values=age_value_max

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Max Market Value vs Age")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [91]:
positions, p_counts = np.unique(epl['position'], return_counts=True)
position_value_mean = []
position_value_median = []
position_value_max = []
for p in positions:
    position_value_mean.append(np.mean(epl['market_value'].loc[epl['position']==p]))
    position_value_median.append(np.median(epl['market_value'].loc[epl['position']==p]))
    position_value_max.append(max(epl['market_value'].loc[epl['position']==p]))

In [84]:
keys= positions
values=position_value_mean

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Mean Market Value vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [85]:
keys= positions
values=position_value_median

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Market Value vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [86]:
keys= positions
values=position_value_max

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Maximum Market Value vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [87]:
position_age_mean = []
position_age_median = []
position_age_max = []
for p in positions:
    position_age_mean.append(np.mean(epl['age'].loc[epl['position']==p]))
    position_age_median.append(np.median(epl['age'].loc[epl['position']==p]))
    position_age_max.append(max(epl['age'].loc[epl['position']==p]))

In [88]:
keys= positions
values=position_age_mean

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Mean Age vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [89]:
keys= positions
values=position_age_median

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Median Age vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [90]:
keys= positions
values=position_age_max

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Max Age vs Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'

In [92]:
keys= positions
values=p_counts

plot({
    "data": [go.Bar(x=keys, y=values)],
    "layout": go.Layout(title="Number of Players in Each Position")
})

'file:///home/arjun/Kaggle/EPL data/temp-plot.html'